src/spatialdata_plot/pl/basic.py

```diff
@@ -459,6 +724,66 @@ def render_labels(
         -------
         None
         """
+        if elements is not None:
+            if not isinstance(elements, (list, str)):
+                raise TypeError("Parameter 'elements' must be a string or a list of strings.")
+
+            elements = [elements] if isinstance(elements, str) else elements
+            if any(e not in self._sdata.labels for e in elements):
+                raise ValueError(
+                    "Not all specificed elements were found, available elements are: '"
+                    + "', '".join(self._sdata.labels.keys())
+                    + "'"
+                )
+
+        if color is not None and not isinstance(color, str):
+            raise TypeError("Parameter 'color' must be a string.")
+
+        if groups is not None:
+            if not isinstance(groups, (list, str)):
+                raise TypeError("Parameter 'groups' must be a string or a list of strings.")
+            groups = [groups] if isinstance(groups, str) else groups
+            if not all(isinstance(g, str) for g in groups):
+                raise TypeError("All items in 'groups' list must be strings.")
+
+        if not isinstance(contour_px, int):
+            raise TypeError("Parameter 'contour_px' must be an integer.")
+
+        if not isinstance(outline, bool):
+            raise TypeError("Parameter 'outline' must be a boolean.")
+
+        if layer is not None and not isinstance(layer, str):
+            raise TypeError("Parameter 'layer' must be a string.")
+
+        if palette is not None:
+            if not isinstance(palette, (list, str)):
+                raise TypeError("Parameter 'palette' must be a string or a list of strings.")
+            palette = [palette] if isinstance(palette, str) else palette
+            if not all(isinstance(p, str) for p in palette):
+                raise TypeError("All items in 'palette' list must be strings.")
+
+        if cmap is not None and not isinstance(cmap, (Colormap, str)):
+            raise TypeError("Parameter 'cmap' must be a Colormap or a string.")
+
+        if norm is not None and not isinstance(norm, Normalize):
+            raise TypeError("Parameter 'norm' must be of type Normalize.")
+
+        if na_color is not None and not colors.is_color_like(na_color):
+            raise ValueError("Parameter 'na_color' must be color-like.")
+
+        if not isinstance(outline_alpha, (float, int)):
+            raise TypeError("Parameter 'outline_alpha' must be numeric.")
+
+        if not isinstance(fill_alpha, (float, int)):
+            raise TypeError("Parameter 'fill_alpha' must be numeric.")
+
+        if scale is not None:
+            if not isinstance(scale, (list, str)):
+                raise TypeError("Parameter 'scale' must be a string or a list of strings.")
+            scale = [scale] if isinstance(scale, str) else scale
+            if not all(isinstance(s, str) for s in scale):
```

Comment:
same as above, None also allowed?

Reply:
Yeah, `None` is allowed. I didn't include it in the error message because it only ever shows when the user meddles with it anyway 🤔